lib/cli.js

```diff
@@ -114,14 +114,24 @@ module.exports = function(program) {
         // So the dash doesn't register as a file
         if (usedDash) { args.length--; }
 
-        checkerPromise = checker.checkStdin().then(function(errors) {
-            return [errors];
-        });
+        if (program.fix) {
+            return {
+                promise: checker.fixStdin().then(function(result) {
+                    process.stdout.write(result.output);
+                }),
+                checker: checker
+            };
+        } else {
```

Comment:
Incorrect structure are hard to parse ;-) Use of `return` makes an usual `if-else` harder to read. That's why I've asked ;-)